src/core/webview/ClineProvider.ts

```diff
@@ -441,7 +442,12 @@ export class ClineProvider
 		// This happens when the user closes the view or when the view is closed programmatically
 		webviewView.onDidDispose(
 			async () => {
-				await this.dispose()
+				if (inTabMode) {
+					this.log("Disposing ClineProvider because it was in tab mode.")
+					await this.dispose()
+				} else {
+					this.log("NOT disposing ClineProvider because it is in sidebar mode and can be reused.")
```

Comment:
Consider improving both log messages for clarity:

```suggestion
				if (inTabMode) {
					this.log("Disposing ClineProvider instance for tab view")
					await this.dispose()
				} else {
					this.log("Preserving ClineProvider instance for sidebar view reuse")
				}
```